src/main/java/com/teragrep/jos_01/procfs/status/Meminfo.java

```diff
@@ -0,0 +1,160 @@
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.jos_01.procfs.status;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+// Provides information about memory usage, measured in pages.
+// Some of these values are inaccurate because of a kernel-internal scalability optimization.
+// If accurate values are required, use smaps or smaps_rollup instead, which are much slower but provide accurate, detailed information
+public class Meminfo implements Status {
+
+    private final ArrayList<String> rows;
+    private final LocalDateTime timestamp;
+    private final Map<String, String> statistics;
+
+    private enum fields {
+        MemTotal,
+        MemFree,
+        MemAvailable,
+        Buffers,
+        Cached,
+        SwapCached,
+        Active,
+        Inactive,
+        Activeanon,
+        Inactiveanon,
+        Activefile,
+        Inactivefile,
+        Unevictable,
+        Mlocked,
+        SwapTotal,
+        SwapFree,
+        Zswap,
+        Zswapped,
+        Dirty,
+        Writeback,
+        AnonPages,
+        Mapped,
+        Shmem,
+        KReclaimable,
+        Slab,
+        SReclaimable,
+        SUnreclaim,
+        KernelStack,
+        PageTables,
+        SecPageTables,
+        NFS_Unstable,
+        Bounce,
+        WritebackTmp,
+        CommitLimit,
+        Committed_AS,
+        VmallocTotal,
+        VmallocUsed,
+        VmallocChunk,
+        Percpu,
+        HardwareCorrupted,
+        AnonHugePages,
+        ShmemHugePages,
+        ShmemPmdMapped,
+        FileHugePages,
+        FilePmdMapped,
+        CmaTotal,
+        CmaFree,
+        HugePages_Total,
+        HugePages_Free,
+        HugePages_Rsvd,
+        HugePages_Surp,
+        Hugepagesize,
+        Hugetlb,
+        DirectMap4k,
+        DirectMap2M,
+        DirectMap1G
+    };
+
+    public Meminfo(ArrayList<String> rows) {
+        this.rows = rows;
+        statistics = new LinkedHashMap<String, String>();
+        Pattern pattern = Pattern.compile("([a-zA-Z0-9_\\(\\)]*): *(\\d*)");
+        for (String row : rows) {
+            Matcher matcher = pattern.matcher(row);
+            if (matcher.find()) {
+                for (int i = 0; i < matcher.groupCount(); i++) {
+                    statistics.put(matcher.group(1), matcher.group(2));
```

Comment:
This sounds like it is adding duplicates? Unless something like this was the intention?
```
while(matcher.find()) {
    statistics.put(matcher.group(1), matcher.group(2));
}
```

Reply:
Removed unnecessary for-loop